### PR TITLE
Rename check_box in checkbox

### LINKED ISSUE #52430: Can we have a checkbox alias?
It is certainly acceptable to refer to checkboxes as two words: "Check Box". But more often, they are referred to as "Checkbox" (one word):
* HTML: [input `type="checkbox"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox)
* Wikipedia: [Checkbox](https://en.wikipedia.org/wiki/Checkbox)
* Nielson Norman Group: [Checkboxes](https://www.nngroup.com/articles/checkboxes-design-guidelines/)

For backwards compatibility, it makes sense to continue supporting `check_box` Form Helper, but I propose adding an alias (or making `check_box` the alias), so that our forms can refer to them as simply "checkbox", without the underscore.

The classes/methods that I think should be aliased are:
* [ActionView::Helpers::FormHelper#check_box](https://github.com/rails/rails/blob/19eebf6d33dd15a0172e3ed2481bec57a89a2404/actionview/lib/action_view/helpers/form_helper.rb#L1343)
* [ActionView::Helpers::FormBuilder#check_box](https://github.com/rails/rails/blob/19eebf6d33dd15a0172e3ed2481bec57a89a2404/actionview/lib/action_view/helpers/form_helper.rb#L2468)
* [ActionView::Helpers::FormBuilder#collection_check_boxes](https://github.com/rails/rails/blob/19eebf6d33dd15a0172e3ed2481bec57a89a2404/actionview/lib/action_view/helpers/form_options_helper.rb#L908)
* [ActionView::Helpers::FormOptionsHelper#collection_check_boxes](https://github.com/rails/rails/blob/19eebf6d33dd15a0172e3ed2481bec57a89a2404/actionview/lib/action_view/helpers/form_options_helper.rb#L782)
* [ActionView::Helpers::FormTagHelper#check_box_tag](https://github.com/rails/rails/blob/19eebf6d33dd15a0172e3ed2481bec57a89a2404/actionview/lib/action_view/helpers/form_tag_helper.rb#L455)
* [ActionView::Helpers::Tags::CheckBox](https://github.com/rails/rails/blob/def0397e595c503099e3c10bd811d5d9e3613b38/actionview/lib/action_view/helpers/tags/check_box.rb)
* [ActionView::Helpers::Tags::CollectionCheckBoxes and CheckBoxBuilder#check_box](https://github.com/rails/rails/blob/def0397e595c503099e3c10bd811d5d9e3613b38/actionview/lib/action_view/helpers/tags/collection_check_boxes.rb)

If this proposal is acceptable, I can draft a PR to make this change. Let me know what you think.

### PR DESCRIPTION
**Pull request by byroot**
But add aliases for backward compatibility.

Fix: https://github.com/rails/rails/issues/52430

### CODE REVIEW

File: actionview/lib/action_view/helpers/form_options_helper.rb [hunk 1]
```diff
@@ -897,16 +897,16 @@ def weekday_select(method, options = {}, html_options = {})
...
       #   <%= form_for @post do |f| %>
-      #     <%= f.collection_check_boxes :author_ids, Author.all, :id, :name_with_initial %>
+      #     <%= f.collection_checkboxes :author_ids, Author.all, :id, :name_with_initial %>
       #     <%= f.submit %>
       #   <% end %>
       #
       # Please refer to the documentation of the base helper for details.
-      def collection_check_boxes(method, collection, value_method, text_method, options = {}, html_options = {}, &block)
-        @template.collection_check_boxes(@object_name, method, collection, value_method, text_method, objectify_options(options), @default_html_options.merge(html_options), &block)
+      def collection_checkboxes(method, collection, value_method, text_method, options = {}, html_options = {}, &block)
```
Comment (chaadow) [inline actionview/lib/action_view/helpers/form_options_helper.rb:908]: hi @byroot I think you forgot to add an alias here as well as in line 782 
(sorry about the screenshot without context but i could reproduce it on our prod app ) 
image

File: actionview/lib/action_view/helpers/form_options_helper.rb [hunk 2]
```diff
@@ -779,7 +779,7 @@ def collection_radio_buttons(object, method, collection, value_method, text_meth
       #
       # In the rare case you don't want this hidden field, you can pass the
       # <tt>include_hidden: false</tt> option to the helper method.
-      def collection_check_boxes(object, method, collection, value_method, text_method, options = {}, html_options = {}, &block)
+      def collection_checkboxes(object, method, collection, value_method, text_method, options = {}, html_options = {}, &block)
```
Comment (chaadow) [inline actionview/lib/action_view/helpers/form_options_helper.rb:782]: here as well

File: actionview/lib/action_view/helpers/form_options_helper.rb (hunk 1 above)
Comment (chaadow) [reply actionview/lib/action_view/helpers/form_options_helper.rb:908]: I opened https://github.com/rails/rails/pull/52436

**Comment by seanpdoyle** [comment]
I've opened https://github.com/rails/rails/pull/52467 in the same spirit of this PR. It renames `text_area_tag` to `textarea_tag` and `rich_text_area_tag` to `rich_textarea_tag`, while preserving backward compatibility with aliases.

**Comment by spickermann** [comment]
I guess it would make sense to deprecate `check_box` and output a warning at some point. And then removing it in a later version. 

Does Ruby on Rails have some kind of guideline or recommendation when to alias or when to deprecate a method? And how long to keep the old version?

**Comment by byroot** [comment]
This isn't something worth deprecating, it's a simple alias.

**Comment by connorgurney** [comment]
This is perhaps one of the most monumental changes to Rails in forever! 😆 Kudos.

**Comment by Earlopain** [comment]
This puts generators in an akward place where they need to handle both variants if they want to properly support both old and new rails versions. https://github.com/rails/tailwindcss-rails/pull/418 Would it make sense to still return the old version from `field_type`?

**Pull request merged**